File: healthcare_app_server/admin_service/src/main/java/dev/skyherobrine/service/models/mongodb/PrivateChat.java
package dev.skyherobrine.service.models.mongodb;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;
import org.springframework.data.mongodb.core.mapping.FieldType;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.time.LocalDateTime;

@Document(collection = "private_chats")
@Getter @Setter @NoArgsConstructor
public class PrivateChat {

    @MongoId @Field(targetType = FieldType.INT64) @NonNull
    private Long id;

    @Field(name = "sender_id", targetType = FieldType.STRING) @NonNull
    private String senderId;

    @Field(name = "receiver_id", targetType = FieldType.STRING) @NonNull
    private String receiverId;

    @Field(targetType = FieldType.STRING) @NonNull
    private String message;

    @Field(name = "is_recall", targetType = FieldType.BOOLEAN) @NonNull
    private Boolean isRecall;

    @Field(name = "is_read", targetType = FieldType.BOOLEAN) @NonNull
    private Boolean isRead;

    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Field(name = "created_at", targetType = FieldType.DATE_TIME) @NonNull
    private LocalDateTime createdAt;

    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Field(name = "updated_at", targetType = FieldType.DATE_TIME) @NonNull
    private LocalDateTime updatedAt;
}

File: healthcare_app_server/chat_service/models/mongodb/private-chat.js
class PrivateChat {
    constructor(id, senderId, receiverId, message) {
        this._is_recall = this._is_read = false;
        this._created_at = this._updated_at = new Date();
        this._id = id;
        this._senderId = senderId;
        this._receiverId = receiverId;
        this._message = message;
    }

    get id() {
        return this._id;
    }

    get senderId() {
        return this._senderId;
    }

    get receiverId() {
        return this._receiverId;
    }

    get message() {
        return this._message;
    }

    get is_recall() {
        return this._is_recall;
    }

    get is_read() {
        return this._is_read;
    }

    get created_at() {
        return this._created_at;
    }

    get updated_at() {
        return this._updated_at;
    }
}